linalg/hypre.hpp

```diff
@@ -967,7 +967,48 @@ class HypreParaSails : public HypreSolver
 
    virtual void SetOperator(const Operator &op);
 
+   /// Set the threshold and levels parameters
+   /** The accuracy and cost of ParaSails are parametrized by the real
+    * @a thresh and integer @a nlevels parameters (0<=thresh<=1,  0<=nlevels).
+    * Lower values of @a thresh and higher values of @a nlevels lead to
+    * more accurate, but more expensive preconditioners. More accurate
+    * preconditioners are also more expensive per iteration. The default
+    * values are @a thresh = 0.1 and @a nlevels = 1.
+    */
+   void SetParameters(double thresh, int nlevels);
+   /// Set the filter parameter
+   /** The filter parameter is used to drop small nonzeros in the preconditioner,
+    * to reduce the cost of applying the preconditioner. Values from 0.055
+    * to 0.1 are recommended. The default value is 0.1.
+    */
+   void SetFilterParameter(double filter);
```

Comment:
Naming: maybe `SetFilter` would be sufficient?

Reply:
Ok, I preferred the word parameter to indicate that no objects are passed, just parameters, but for consistency with hypre it is maybe better without it.